modules/build/src/main/scala/scala/build/internal/NativeBuilderHelper.scala

```diff
@@ -0,0 +1,65 @@
+package scala.build.internal
+
+import java.math.BigInteger
+import java.security.MessageDigest
+
+import scala.build.Build
+import scala.scalanative.{build => sn}
+
+case object NativeBuilderHelper {
+
+  private def resolveProjectShaPath(nativeWorkDir: os.Path) = nativeWorkDir / ".project_sha"
+  private def resolveOutputShaPath(nativeWorkDir: os.Path)  = nativeWorkDir / ".output_sha"
+
+  private def fileSha(filePath: os.Path): String = {
+    val md = MessageDigest.getInstance("SHA-1")
+    md.update(os.read.bytes(filePath))
+
+    val digest        = md.digest()
+    val calculatedSum = new BigInteger(1, digest)
+    String.format(s"%040x", calculatedSum)
+  }
+
+  private def projectSha(build: Build.Successful, nativeConfig: sn.NativeConfig) = {
+    val md = MessageDigest.getInstance("SHA-1")
+    md.update(build.inputs.sourceHash().getBytes)
+    md.update(nativeConfig.toString.getBytes)
```

Comment:
We shouldn't need to hash the native config here, as it originates from the build options (that are hashed right after). I opened https://github.com/VirtusLab/scala-cli/pull/394 to make that more explicit.

Reply:
I think in some cases, it's required to count hash for native config.  For example, if clang path was changed, we do not detect this change in build options, because we discover it after creating buildOptions -> [discover clang path](https://github.com/VirtusLab/scala-cli/blob/master/modules/build/src/main/scala/scala/build/options/ScalaNativeOptions.scala#L39)